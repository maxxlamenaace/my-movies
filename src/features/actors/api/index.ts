import { getActorDetails } from './get-actor-details/get-actor-details';
import { getActorCombinedCredits } from './get-actor-combined-credits/get-actor-combined-credits';

export { getActorDetails, getActorCombinedCredits };
