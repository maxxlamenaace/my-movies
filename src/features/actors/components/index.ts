import ActorsGrid from './actors-grid/ActorsGrid';
import ActorItem from './actor-item/ActorItem';
import ActorInfos from './actor-infos/ActorInfos';

export { ActorsGrid, ActorItem, ActorInfos };
