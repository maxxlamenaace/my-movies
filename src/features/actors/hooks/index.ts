import { useActorDetails } from './use-actor-details/use-actor-details';

export { useActorDetails };
