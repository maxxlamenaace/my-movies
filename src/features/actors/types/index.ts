import { Media } from '@/features/medias';

export type Actor = {
  id: number;
  birthday: string;
  known_for_department: string;
  deathday?: string;
  name: string;
  also_known_as: string[];
  gender: number;
  biography: string;
  popularity: number;
  place_of_birth: string;
  profile_path: string;
  adult: boolean;
  imdb_id: string;
  homepage: string;
};

export type ActorImage = {
  id: number;
  profiles: ActorImageProfile[];
};

export type ActorImageProfile = {
  aspect_ratio: number;
  file_path: string;
  height: number;
  width: number;
};

export type ActorDetails = {
  credits: Media[];
} & Actor;
