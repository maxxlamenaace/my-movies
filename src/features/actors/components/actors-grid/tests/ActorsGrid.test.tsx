import React from 'react';

import { BrowserRouter } from 'react-router-dom';
import { render, cleanup } from '@testing-library/react';

import { actor } from '@/tests/factories';
import { ActorsGrid } from '@/features/actors';

describe('ActorsGrid', () => {
  afterEach(() => {
    cleanup();
  });

  it('renders correctly', () => {
    const { asFragment } = render(<ActorsGrid actors={[actor()]} />, {
      wrapper: BrowserRouter,
    });

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with as many actor item as needed', () => {
    const { container } = render(<ActorsGrid actors={[actor()]} />, {
      wrapper: BrowserRouter,
    });

    expect(container.querySelectorAll('.actor-grid-item')).toHaveLength(1);
  });
});
