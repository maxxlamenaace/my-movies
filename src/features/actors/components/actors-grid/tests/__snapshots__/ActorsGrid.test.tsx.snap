// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActorsGrid renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-1drqt17-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4 MuiGrid-grid-md-3 actor-grid-item css-1vm8tgf-MuiGrid-root"
    >
      <a
        class="actor-item"
        href="/actors/1230166331"
      >
        <div
          class="MuiBox-root css-itb0du"
        >
          <div
            class="media-back-drop MuiBox-root css-2e5d9x"
          />
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium media-play-btn css-3i9v1t-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-17slstj-MuiSvgIcon-root"
              data-testid="PersonSearchIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <circle
                cx="10"
                cy="8"
                r="4"
              />
              <path
                d="M10.35 14.01C7.62 13.91 2 15.27 2 18v2h9.54c-2.47-2.76-1.23-5.89-1.19-5.99zm9.08 4.01c.36-.59.57-1.28.57-2.02 0-2.21-1.79-4-4-4s-4 1.79-4 4 1.79 4 4 4c.74 0 1.43-.22 2.02-.57L20.59 22 22 20.59l-2.57-2.57zM16 18c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div
            class="media-info MuiBox-root css-5bs0yh"
          >
            <div
              class="css-1okj3ks-MuiStack-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-qcvrn6-MuiTypography-root"
              >
                name
              </p>
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
